inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -308,7 +308,13 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
         // Remove Reorder for Convolution: b_fs_yx_fsv32 (i8/u8) -> b_fs_yx_fsv16 (fp32/fp16)
         if (next.is_type<convolution>() && fmt_prev == format::b_fs_yx_fsv32 && fmt_next == format::b_fs_yx_fsv16 &&
             !data_type_traits::is_floating_point(prev_dt) && data_type_traits::is_floating_point(next_dt)) {
-            return true;
+            auto& node = prev.get_users().front();
+            // Avoid to fuse padding reorder to previous onednn convolution
+            if (prev.is_type<convolution>() && prev.as<convolution>().get_preferred_impl_type() == impl_types::onednn &&
+                (node->get_output_layout().data_padding != prev.get_output_layout().data_padding))
```

Comment:
What about using `are_data_types_suitable_for_onednn` function here?

Reply:
To be followed up: it is not only for conv, but also for other onednn primitives as well.

Reply:
This will be applied to next PR with general reorder removal together.